runelite-client/src/main/java/net/runelite/client/plugins/cerberus/CerberusOverlay.java

```diff
@@ -24,53 +25,108 @@
  */
 package net.runelite.client.plugins.cerberus;
 
+import java.awt.BasicStroke;
+import java.awt.Color;
 import java.awt.Dimension;
 import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
 import javax.inject.Inject;
-import javax.inject.Singleton;
+import net.runelite.api.Client;
+import net.runelite.api.NPC;
+import net.runelite.api.NpcID;
+import net.runelite.api.Perspective;
+import net.runelite.api.Point;
+import net.runelite.api.Skill;
+import net.runelite.api.coords.LocalPoint;
 import net.runelite.client.game.SkillIconManager;
 import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
 import net.runelite.client.ui.overlay.OverlayPosition;
-import net.runelite.client.ui.overlay.components.ImageComponent;
-import net.runelite.client.ui.overlay.components.PanelComponent;
 
-@Singleton
 public class CerberusOverlay extends Overlay
 {
-	private final CerberusPlugin plugin;
-	private final SkillIconManager iconManager;
-	private final PanelComponent panelComponent = new PanelComponent();
+	private static final Color COLOR_ICON_BACKGROUND = new Color(0, 0, 0, 128);
+	private static final Color COLOR_ICON_BORDER = new Color(0, 0, 0, 255);
+	private static final int OVERLAY_ICON_DISTANCE = 50;
+	private static final int OVERLAY_ICON_MARGIN = 8;
+	private static final int OVERLAY_Z_OFFSET = 100;
+
+	private Client client;
+	private CerberusPlugin plugin;
+
+	@Inject
+	private SkillIconManager iconManager;
 
 	@Inject
-	CerberusOverlay(final CerberusPlugin plugin, final SkillIconManager iconManager)
+	public CerberusOverlay(Client client, CerberusPlugin plugin)
 	{
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.ABOVE_SCENE);
+		this.client = client;
 		this.plugin = plugin;
-		this.iconManager = iconManager;
-		setPosition(OverlayPosition.BOTTOM_RIGHT);
-		panelComponent.setOrientation(PanelComponent.Orientation.HORIZONTAL);
 	}
 
+	private BufferedImage getIcon(int ghostType)
+	{
+		switch (ghostType)
+		{
+			case NpcID.SUMMONED_SOUL: return iconManager.getSkillImage(Skill.RANGED);
+			case NpcID.SUMMONED_SOUL_5868: return iconManager.getSkillImage(Skill.MAGIC);
+			case NpcID.SUMMONED_SOUL_5869: return iconManager.getSkillImage(Skill.ATTACK);
+		}
+		return null;
+	}
+
+
 	@Override
 	public Dimension render(Graphics2D graphics)
 	{
-		if (plugin.getGhosts().isEmpty())
+		for (NPC ghost : plugin.getGhosts())
 		{
-			return null;
-		}
+			LocalPoint lp = ghost.getLocalLocation();
+			if (lp != null)
+			{
+				Point point = Perspective.localToCanvas(client, lp, client.getPlane(),
+					OVERLAY_Z_OFFSET);
+
+				if (point != null)
+				{
+					point = new Point(point.getX(), point.getY());
+
+					BufferedImage icon = getIcon(ghost.getId());
+
+					if (icon == null)
+					{
+						continue;
+					}
 
-		panelComponent.getChildren().clear();
+					int width = OVERLAY_ICON_MARGIN + icon.getWidth();
```

Comment:
This part of the render loop might be easier to read if it were moved into its own function, perhaps `private static void renderPrayerIndicator(Graphics2D graphics, BufferedImage icon, Point point)`?